index/scorch/merge.go

```diff
@@ -129,8 +147,79 @@ func (s *Scorch) parseMergePlannerOptions() (*mergeplan.MergePlanOptions,
 	return &mergePlannerOptions, nil
 }
 
+type mergerCtrl struct {
+	creator string
+	options *mergeplan.MergePlanOptions
+	doneCh  chan struct{}
+}
+
+func (s *Scorch) RequestMerge(mo *mergeplan.MergePlanOptions,
+	doneCh chan struct{}) error {
+	// check whether force merge is already under processing
+	if atomic.LoadUint64(&s.stats.TotFileMergeForceOpsStarted) >
+		atomic.LoadUint64(&s.stats.TotFileMergeForceOpsCompleted) {
+		if doneCh != nil {
+			close(doneCh)
+		}
+		return fmt.Errorf("compaction already in progress")
+	}
+
+	go func() {
+		defer func() {
+			// doneCh for the caller to track the async merge completion
+			if doneCh != nil {
+				close(doneCh)
+			}
+		}()
+
+		if atomic.LoadUint64(&s.stats.TotFileMergeForceOpsStarted) >
+			atomic.LoadUint64(&s.stats.TotFileMergeForceOpsCompleted) {
+			return
```

Comment:
this is the second time we've checked the stats to see if it's already running, however in the first case you get a helpful error message, and in this case you don't (yes I see the goroutine has no return, but i'm wondering what the thinking is for this difference in behavior)

Reply:
good catch! i missed it..